Src/RandomizerTMF.Logic/RandomizerEngine.cs

```diff
@@ -849,6 +851,19 @@ private static async Task PrepareNewMapAsync(CancellationToken cancellationToken
     {
         Status("Fetching random track...");
 
+        if (Config.Rules.EvenEnvironmentDistribution)
+        {
+            var (site, envi) = GetRandomEEnvironment();
+            Config.Rules.RequestRules.Site = site;
```

Comment:
I think integrating this functionality to `ToUrl()` would be ideal, instead of directly messing with the Config object. Maybe adding some parameter to `ToUrl()`, like `RandomizationRules additionalRules` where inside of `ToUrl()` the `if`s would go. Once this is resolved, I am good to merge.

Also, you don't need to do `new Random()` anymore, there's `Random.Shared` that is used within the project instead (currently, maybe in the future some secure randomizing methods could be added).

Reply:
Maybe the `Even...` properties could be put into `RequestRules` directly and then for example **above** this line https://github.com/BigBang1112/randomizer-tmf/blob/4f43f2b6389c2b11a9336ccafd9a63b1e84c2708/Src/RandomizerTMF.Logic/RequestRules.cs#L77
Have an if check like ```cs
if (EvenEnvironmentDistribution)
{
	if (prop.Name == nameof(Vehicle))
    {
        val = `GetRandomEnv()` // pure idea example
    }
}
```
I can add the validation rules (so that TMNF is disallowed for example) myself.

Reply:
Oh well, you also need to make sure the iteration doesn't get skipped immediately in the first condition of the `foreach`.